File: test/java/LIM2Metrics/CohesionMetrics/LCOM5_AccessesAttribute/LCOM5_AccessesAttribute.java
package regtest;

class SuperBase{
  public int superBaseVar;
}

class Base extends SuperBase{
  public int baseVar1;
}

class ClassType{
  public int classTypeVar;
}

//LCOM5 = 5
class LCOM5_AccessesAttribute extends Base{
  public int localVar;
  public ClassType localObj;
    
  public void foo_localVar(){
    localVar++;
  }

  public void goo_localVar(){
    localVar++;
  }

  public void foo_baseVar(){
    baseVar1++;
  }

  public void goo_baseVar(){
    baseVar1++;
  }

  public void foo_superBaseVar(){
    superBaseVar++;
  }

  public void goo_superBaseVar(){
    superBaseVar++;
  }

  public void foo_objVarAccess(){
    localObj.classTypeVar++;
  }

  public void goo_objVarAccess(){
    localObj.classTypeVar++;
  }

  public static void main(String[] args){}
}

File: test/cpp/LIM2Metrics/CohesionMetrics/LCOM5_AccessesAttribute/LCOM5_AccessesAttribute.cpp
class SuperBase{
public:
    int superBaseVar;
};
    
class Base : public SuperBase{
public:
    int baseVar1;
};

class ClassType{
public:
    int classTypeVar;
};

//LCOM5 = 4
class MyClass : public Base{
    int localVar;
    ClassType localObj;
    
    void foo_localVar(){
        localVar;
    }
    
    void goo_localVar(){
        localVar;
    }
    
    void foo_baseVar(){
        baseVar1;
    }
    
    void goo_baseVar(){
        baseVar1;
    }
    
    void foo_superBaseVar(){
        superBaseVar;
    }
    
    void goo_superBaseVar(){
        superBaseVar;
    }
    
    void foo_objVarAccess(){
        localObj.classTypeVar;
    }
    
    void goo_objVarAccess(){
        localObj.classTypeVar;
    }

};    

int main(){
    return 0;
}